### PR TITLE
openshift/assisted-service: Add e2e workflow

### PR DESCRIPTION
**Pull request by lranjbar**
[MGMT-3324](https://issues.redhat.com/browse/MGMT-3324)

### CODE REVIEW

**Comment by YuviGold** [comment]
Please import `assisted-service` ImageStream and use it in the `e2e-metal-assisted-*` periodics, like being done with `assisted-installer`
* https://github.com/openshift/release/blob/master/ci-operator/config/openshift/release/openshift-release-master__ocp-4.6.yaml
* https://github.com/openshift/release/blob/master/ci-operator/config/openshift/release/openshift-release-master__ocp-4.7.yaml

File: ci-operator/config/openshift/release/openshift-release-master__ocp-4.6.yaml [hunk 1]
```diff
@@ -15,6 +15,10 @@ base_images:
     name: "4.7"
     namespace: ocp
     tag: assisted-installer-controller
+  assisted-service:
+    name: "4.7"
+    namespace: ""
```
Comment (YuviGold) [inline ci-operator/config/openshift/release/openshift-release-master__ocp-4.6.yaml:20]: ocp

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml [hunk 2]
```diff
@@ -32,12 +37,15 @@ tag_specification:
...
 test_binary_build_commands: |
   make build-openshift-ci-test-bin
+  curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.24.0
+  go mod vendor
+  mkdir /.cache && chmod 775 -R /.cache && chmod 775 -R $GOPATH
 tests:
 - artifact_dir: /tmp/artifacts
   as: lint
-  commands: export GOLANGCI_LINT_CACHE=/tmp/.cache make lint
+  commands: export GOLANGCI_LINT_CACHE=/tmp/.cache && export GOPROXY=https://proxy.golang.org && make lint
```
Comment (YuviGold) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:46]: I think this code should be devided to another PR.

File: ci-operator/config/openshift/release/openshift-release-master__ocp-4.6.yaml (hunk 1 above)
Comment (lranjbar) [reply ci-operator/config/openshift/release/openshift-release-master__ocp-4.6.yaml:20]: This is the output of using `make ci-operator-config` and the jobs will fail with the ocp namespace here.

Comment (YuviGold) [reply ci-operator/config/openshift/release/openshift-release-master__ocp-4.6.yaml:20]: The promotion is defined under the assisted-service configuration for "ocp/4.7". No way it would work that way with an empty namespace.

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml [hunk 3]
```diff
@@ -55,6 +63,27 @@ tests:
...
+- artifact_dir: /tmp/artifacts
+  as: e2e-metal-assisted
+  steps:
+    cluster_profile: packet
+    dependencies:
+      ASSISTED_SERVICE_IMAGE: pipeline:assisted-service
+    env:
+      ASSISTED_CONFIG: |
+        MAKEFILE_TARGET="create_full_environment run"
+    workflow: baremetalds-assisted
```
Comment (YuviGold) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:75]: Please switch to workflow `baremetalds-assisted-workflow` as introduced on https://github.com/openshift/release/pull/15647 to ease the PRs process.

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml [hunk 4]
```diff
@@ -55,6 +63,27 @@ tests:
...
     workflow: ipi-aws
+- artifact_dir: /tmp/artifacts
+  as: e2e-metal-assisted
+  steps:
+    cluster_profile: packet
+    dependencies:
+      ASSISTED_SERVICE_IMAGE: pipeline:assisted-service
+    env:
+      ASSISTED_CONFIG: |
+        MAKEFILE_TARGET="create_full_environment run"
```
Comment (YuviGold) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:74]: Why did you override the default Makefile target? we would like to verify a full installation.

File: ci-operator/config/openshift/release/openshift-release-master__ocp-4.6.yaml (hunk 1 above)
Comment (YuviGold) [reply ci-operator/config/openshift/release/openshift-release-master__ocp-4.6.yaml:20]: Sorry, I saw all the namespaces got bumped on https://github.com/openshift/release/pull/15597
You should change it accordingly to "4.8"

Comment (YuviGold) [reply ci-operator/config/openshift/release/openshift-release-master__ocp-4.6.yaml:20]: Ok I made some changes on https://github.com/openshift/release/pull/15742 to avoid those numbers.
We should use `ocp-kni` namespace and `assisted` repository

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml (hunk 4 above)
Comment (lranjbar) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:74]: I will test this workflow but I have been testing a simplified flow for these jobs that doesn't time out.

**Comment by lranjbar** [comment]
@YuviGold This is still heavily a work in progress and there are several issues I'm working through still.

**Comment by lranjbar** [comment]
@djzager @YuviGold So this one is good for review finally.

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml (hunk 2 above)
Comment (djzager) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:50]: I can see why it could be done in another PR but that isn't too important to me.

**Review by djzager — APPROVED** [review]
/lgtm

**Review by YuviGold — APPROVED** [review]
LGTM

**Comment by lranjbar** [comment]
Just rebased to resolve conflicts /unhold

**Comment by lranjbar** [comment]
/unhold /assign @droslean

Rebase is complete now and we need an approval for the periodics

**Review by droslean — APPROVED** [review]
